### DISCUSSION TITLE
Applying Diátaxis to course structure

### DISCUSSION THREAD

**Discussion by honzajavorek** [evildmp/diataxis-documentation-framework#130] [Q&A]
Hi, I struggle to apply Diátaxis to a course structure. I don't have much practice applying Diátaxis in general, so it's possible I just don't understand it properly, but it just feels that course is a different beast than docs.

Let's say I define course as a collection of lessons. People take the course to learn something. Lessons guide the pupil through the learning. Imagine a course like "Intro to Web Scraping" or "Advanced GraphQL APIs", not something necessarily tied to a certain product.

It kind of feels like a giant tutorial, or perhaps a collection of tutorials. However, it also feels strange to expel all explanation out of the lessons. When going through an online course or through a learning book, explaining the context is usually a large portion of the content.

What would those more experienced in applying Diátaxis suggest? My idea would be e.g. to have a course as a collection of lessons, where one lesson consists of three parts: explanation intro, then the actual tutorial, then perhaps ending relevant links to further info (further explanation, reference, howtos), but those must not compete visually with that big _next lesson_ button.

What do you, fellow Diátaxis fans, think? Any good resources on this topic?

**Comment by drvinceknight** [comment]
I teach a first year programming University course using a text written based on the Diataxis framework.

The text I use is here: https://vknight.org/pfm/cover.html

In general every topic covered is done according to the following:

1. I demo a tutorial and then ask the students to do the tutorial.
2. I give the students tasks and encourage them to use the how to guides to complete them.
3. I discuss the topic in more depth using the explanation section.
4. The reference section is available to them for more information.

In the later part of the course I actually teach the Diataxis framework itself (when we cover the topic of documentation).

I'd be keen to hear others' thoughts and experiences :)

**Reply by honzajavorek** [reply]
Wow, that's awesome. I like how you consistently use the structure (tutorial, how, exercises, solutions, further info) and make it transparent to the reader. When you're referring to the explanation section, which one do you mean? In your mind, is further info explanation or reference?

**Comment by drvinceknight** [comment]
In my mind (for the purpose of this course) further information is the explanation section. For example [How does Python sample randomness?](https://vknight.org/pfm/tools-for-mathematics/06-probability/why/main.html#how-does-python-sample-randomness) is in the "Further Information" section. I don't expect my students here to all **need** to know how randomness works but some do **want** to.
